clients/rust/tests/initialize_config.rs

```diff
@@ -0,0 +1,581 @@
+            &context.payer.pubkey(),
+            &config.pubkey(),
+            context
+                .banks_client
+                .get_rent()
+                .await
+                .unwrap()
+                .minimum_balance(Config::LEN),
+            Config::LEN as u64,
+            &paladin_stake::ID,
+        );
+
+        let intialize_ix = InitializeConfigBuilder::new()
+            .config(config.pubkey())
+            .config_authority(authority_pubkey)
+            .slash_authority(authority_pubkey)
+            .mint(mint.pubkey())
+            .vault(token.pubkey())
+            .cooldown_time_seconds(1) // 1 second
+            .max_deactivation_basis_points(500) // 5%
+            .instruction();
+
+        // When we try to initialize the config with the mint without a transfer hook.
+
+        let tx = Transaction::new_signed_with_payer(
+            &[create_ix, intialize_ix],
+            Some(&context.payer.pubkey()),
+            &[&context.payer, &config],
+            context.last_blockhash,
+        );
+
+        let err = context
+            .banks_client
+            .process_transaction(tx)
+            .await
+            .unwrap_err();
+
+        // Then we expect an error.
+
+        assert_custom_error!(err, StakeError::MissingTransferHook);
+    }
+
+    #[tokio::test]
+    async fn fail_initialize_config_with_unitialized_mint() {
+        let mut context = ProgramTest::new("stake_program", paladin_stake::ID, None)
+            .start_with_context()
+            .await;
+
+        // Given an empty config account and a mint.
+
+        let config = Keypair::new();
+        let authority = Keypair::new().pubkey();
+
+        let mint = Keypair::new();
+        let token = Keypair::new();
+        let rent = context.banks_client.get_rent().await.unwrap();
+
+        let create_mint_ix = system_instruction::create_account(
+            &context.payer.pubkey(),
+            &mint.pubkey(),
+            rent.minimum_balance(Config::LEN),
+            Config::LEN as u64,
+            &spl_token_2022::ID,
+        );
+
+        let create_ix = system_instruction::create_account(
+            &context.payer.pubkey(),
+            &config.pubkey(),
+            context
+                .banks_client
+                .get_rent()
+                .await
+                .unwrap()
+                .minimum_balance(Config::LEN),
+            Config::LEN as u64,
+            &paladin_stake::ID,
+        );
+
+        let initialize_ix = InitializeConfigBuilder::new()
+            .config(config.pubkey())
+            .config_authority(authority)
+            .slash_authority(authority)
+            .mint(mint.pubkey())
+            .vault(token.pubkey())
+            .cooldown_time_seconds(1) // 1 second
+            .max_deactivation_basis_points(500) // 5%
+            .instruction();
+
+        // When we try to initialize the config with the wrong token authority.
```

Comment:
micro-nit: Should probably say "When we try to initialize the config with an uninitialized mint.", right?

Reply:
Yep, copy-paste left over.